### PR TITLE
feat: version 3.1.0

### PR DESCRIPTION
**Pull request by suhailkakar**
- replaces `axios` with native `fetch` for calling APIs
- introduces Zod[2] to validate user input and api response
- works natively across browser, node, or any other js runtime (like react native, bun)
- tree-shaking import, users can import methods they want to use, and only those methods will be included in a bundle not the entire SDK

### CODE REVIEW

File: .speakeasy/gen.yaml [hunk 1]
```diff
@@ -1,21 +1,23 @@
-configVersion: 1.0.0
+configVersion: 2.0.0
 generation:
-  comments: {}
```
Comment (0xcadams) [inline .speakeasy/gen.yaml:3]: Would this be a breaking change? We may want to bump version with a breaking release if so

File: .speakeasy/gen.yaml [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
-    globalServerURLs: 2.82.0
-    inputOutputModels: 2.82.0
-    nameOverrides: 2.81.1
+  useClassNamesForArrayFields: true
+  fixes:
+    nameResolutionDec2023: true
+    parameterOrderingFeb2024: true
+    requestResponseComponentNamesFeb2024: true
+  auth:
+    oAuth2ClientCredentialsEnabled: false
```
Comment (0xcadams) [inline .speakeasy/gen.yaml:13]: What does this do?

File: .speakeasy/gen.yaml [hunk 3]
```diff
@@ -1,21 +1,23 @@
...
 typescript:
-  version: 3.0.2
-  author: Livepeer
+  version: 0.0.2
```
Comment (0xcadams) [inline .speakeasy/gen.yaml:15]: Is this the right version?

File: .speakeasy/gen.yaml [hunk 4]
```diff
@@ -1,21 +1,23 @@
...
+  additionalDependencies:
+    dependencies: {}
+    devDependencies: {}
+    peerDependencies: {}
+  author: Speakeasy
```
Comment (0xcadams) [inline .speakeasy/gen.yaml:20]: Do we want to publish the author as Speakeasy?

File: README.md [hunk 5]
```diff
@@ -1,277 +1,417 @@
...
+    <a href="https://opensource.org/licenses/MIT">
+        <img src="https://img.shields.io/badge/License-MIT-blue.svg" style="width: 100px; height: 28px;" />
+    </a>
+</div>
 
-The Livepeer JavaScript library provides convenient access to the Livepeer Studio API from
-applications written in both browser and server-side JavaScript
 
-## Documentation
+## 🏗 **Welcome to your new SDK!** 🏗
```
Comment (0xcadams) [inline README.md:11]: Looks like these docs need updating

File: RUNTIMES.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Supported JavaScript runtimes
```
Comment (0xcadams) [inline RUNTIMES.md:1]: Do we know if this means it's actually not compatible with edge runtimes? 🤔

Comment (0xcadams) [reply RUNTIMES.md:1]: I don't see any APIs here which aren't supported in every major environment type, like edge and those.

File: src/models/components/asset.ts [hunk 7]
```diff
@@ -119,288 +148,1173 @@ export enum AssetPhase {
...
-    @Expose({ name: "bitDepth" })
-    bitDepth?: number;
-}
+    bitDepth?: number | undefined;
+};
 
 /**
  * Video metadata
  */
-export class VideoSpec extends SpeakeasyBase {
```
Comment (0xcadams) [inline src/models/components/asset.ts:269]: I really like that this is not classes anymore - those were a pain to work with in RSC! This Zod schema generation is magical.

**Review by 0xcadams — COMMENTED** [review]
Just a few changes here for docs that may have regressed with the new auto-generation run.

**Comment by 0xcadams** [comment]
Hey Suhail - do you need anything from me to get this landed?

**Comment by suhailkakar** [comment]
Closing this PR and opened #5 instead with new speakeasy workflow and adding tests!

**Pull request closed without merging**